### PR TITLE
extra params for tools (mainly Sonar)

### PR DESCRIPTION
**Pull request by sergehog**

### CODE REVIEW

File: hallux/auxilary.py [hunk 1]
```diff
@@ -29,10 +29,25 @@ def set_directory(path: str | Path | TemporaryDirectory):
...
+def find_argvalue(argv: list[str], name: str) -> str | None:
+    """
+    Finds argument index and following value
+    """
+    for i in range(len(argv)):
+        arg = argv[i]
+        if arg == name:  # --key value
+            if len(argv) > i + 1 and not argv[i + 1].startswith("-"):
+                return argv[i + 1]
+        elif arg.startswith(name + "="):  # --key=value
```
Comment (HalluxAIDev) [inline hallux/auxilary.py:50]: Remove this commented out code.
```suggestion
    for i in range(len(argv)):
        arg = argv[i]
        if arg == name:
            if len(argv) > i + 1 and not argv[i + 1].startswith("-"):
                return argv[i + 1]
        elif arg.startswith(name + "="):

```

File: hallux/targets/github_proposal.py [hunk 2]
```diff
@@ -120,9 +121,10 @@ def commit_diff(self) -> bool:
                 side="RIGHT",
                 path=str(compacted.filename),
                 line=compacted.end_line,
-                start_line=compacted.start_line,
+                start_line=compacted.start_line if compacted.start_line > compacted.end_line else GithubObject.NotSet,
             )
-        except GithubException:
+        except BaseException as ex:
```
Comment (HalluxAIDev) [inline hallux/targets/github_proposal.py:126]: Catch a more specific exception or reraise the exception
```suggestion
        except Exception as ex: 

```

File: hallux/auxilary.py (hunk 1 above)
Comment (sergehog) [inline hallux/auxilary.py:50]:
```suggestion
        elif arg.startswith(name + "="):
```

File: hallux/auxilary.py [hunk 3]
```diff
@@ -29,10 +29,25 @@ def set_directory(path: str | Path | TemporaryDirectory):
...
         if arg == name or arg.startswith(name):
             return i
     return -1
+
+
+def find_argvalue(argv: list[str], name: str) -> str | None:
+    """
+    Finds argument index and following value
+    """
+    for i in range(len(argv)):
```
Comment (sergehog) [inline hallux/auxilary.py:45]: Remove this commented out code.
```suggestion
    for i in range(len(argv)):
        arg = argv[i]
        if arg == name:
            if len(argv) > i + 1 and not argv[i + 1].startswith("-"):
                return argv[i + 1]
        elif arg.startswith(name + "="):

```

File: hallux/auxilary.py (hunk 1 above)
Comment (sergehog) [inline hallux/auxilary.py:50]:
```suggestion
    for i in range(len(argv)):
        arg = argv[i]
        if arg == name:
            if len(argv) > i + 1 and not argv[i + 1].startswith("-"):
                return argv[i + 1]
        elif arg.startswith(name + "="):
```

File: hallux/targets/github_proposal.py [hunk 4]
```diff
@@ -118,13 +117,37 @@ def commit_diff(self) -> bool:
...
+        #         line=compacted.end_line,
+        #         start_line=compacted.start_line if compacted.start_line > compacted.end_line else GithubObject.NotSet,
+        #     )
+        # except BaseException as ex:
+        #     logger.debug(ex)
+        #     success = False
+
         # clean local code
         FilesystemTarget.revert_diff(self)
         return success
```
Comment (HalluxAIDev) [inline hallux/targets/github_proposal.py:153]: Remove this commented out code.
```suggestion
    def commit_diff(self) -> bool:
        commit = self.repo.get_commit(self.pull_request.head.sha)
        body = self.existing_proposal.description + "\n```suggestion\n"

        for line in self.existing_proposal.proposed_lines:
            body = body + line + "\n"
        body = body + "\n```"

        success: bool = True
        try:
            self.pull_request.create_review_comment(
                body=body,
                commit=commit,
                side="RIGHT",
                path=str(self.existing_proposal.filename),
                line=self.existing_proposal.end_line,
                start_line=self.existing_proposal.start_line
                if self.existing_proposal.start_line > self.existing_proposal.end_line
                else GithubObject.NotSet,
            )
        except BaseException as ex:
            logger.debug(ex)
            success = False

        FilesystemTarget.revert_diff(self)
        return success

```

File: hallux/auxilary.py [hunk 5]
```diff
@@ -29,10 +29,25 @@ def set_directory(path: str | Path | TemporaryDirectory):
...
+        arg = argv[i]
+        if arg == name:  # --key value
+            if len(argv) > i + 1 and not argv[i + 1].startswith("-"):
+                return argv[i + 1]
+        elif arg.startswith(name + "="):  # --key=value
+            return arg[len(name) + 1 :]
+
+    return None
```
Comment (HalluxAIDev) [inline hallux/auxilary.py:53]: Remove this commented out code.
```suggestion
def find_argvalue(argv: list[str], name: str) -> str | None:
    for i in range(len(argv)):
        arg = argv[i]
        if arg == name:
            if len(argv) > i + 1 and not argv[i + 1].startswith("-"):
                return argv[i + 1]
        elif arg.startswith(name + "="):
            return arg[len(name) + 1 :]

    return None

```

File: hallux/tools/sonarqube/solver.py [hunk 6]
```diff
@@ -58,6 +72,8 @@ def __init__(
         self.token: Final[str | None] = token
         self.url: Final[str | None] = url
         self.project: Final[str | None] = project
+        self.search_params: Final[str] = search_params
+        self.extra_param: Final[str | None] = extra_param
 
         # With SonarQube we only may solve 1 issue per file.
         self.already_fixed_files: Final[list[str]] = []
```
Comment (HalluxAIDev) [inline hallux/tools/sonarqube/solver.py:79]: Merge this if statement with the enclosing one.
```suggestion
    def __init__(
        self,
        config_path: Path,
        run_path: Path,
        command_dir: str = ".",
        validity_test: str | None = None,
        url: str | None = None,
        token: str | None = None,
        project: str | None = None,
        search_params: str = "resolved=false&severities=MINOR,MAJOR,CRITICAL&statuses=OPEN,CONFIRMED,REOPENED",
        extra_param: str | None = None,  # example: "branch=65-sonar-fixes"
    ):
        """
        IssueSolver implementation for SonarQube
        :param config_path: Directory with .hallux file
        :param run_path: Directory, where hallux was run
        :param command_dir: Directory, passed to hallux in CLI
        :param validity_test: script
        :param url:
        :param token:
        :param project:
        :param search_params:
        :param extra_param:
        """
        super().__init__(config_path, run_path, command_dir, validity_test=validity_test)

        self.token: Final[str | None] = token if token is not None else os.getenv(self.SONAR_TOKEN)
        self.url: Final[str | None] = url
        self.project: Final[str | None] = project
        self.search_params: Final[str] = search_params
        self.extra_param: Final[str | None] = extra_param

        # With SonarQube we only may solve 1 issue per file.
        self.already_fixed_files: Final[list[str]] = []

```

File: hallux/targets/github_proposal.py [hunk 7]
```diff
@@ -118,13 +117,37 @@ def commit_diff(self) -> bool:
...
+                start_line=self.existing_proposal.start_line
+                if self.existing_proposal.start_line > self.existing_proposal.end_line
+                else GithubObject.NotSet,
             )
-        except GithubException:
+        except BaseException as ex:
+            logger.debug(ex)
             success = False
 
+        # compacted = self.compact_proposal(self.existing_proposal)
```
Comment (HalluxAIDev) [inline hallux/targets/github_proposal.py:130]: Remove this commented out code.
```suggestion
        FilesystemTarget.revert_diff(self)

```

File: hallux/auxilary.py [hunk 8]
```diff
@@ -29,10 +29,25 @@ def set_directory(path: str | Path | TemporaryDirectory):
...
+
+def find_argvalue(argv: list[str], name: str) -> str | None:
+    """
+    Finds argument index and following value
+    """
+    for i in range(len(argv)):
+        arg = argv[i]
+        if arg == name:  # --key value
+            if len(argv) > i + 1 and not argv[i + 1].startswith("-"):
+                return argv[i + 1]
```
Comment (HalluxAIDev) [inline hallux/auxilary.py:49]: Remove this commented out code.
```suggestion
    for i in range(len(argv)):
        arg = argv[i]
        if arg == name:
            if len(argv) > i + 1 and not argv[i + 1].startswith("-"):
                return argv[i + 1]
        elif arg.startswith(name + "="):

```

File: hallux/targets/github_proposal.py [hunk 9]
```diff
@@ -119,10 +119,14 @@ def commit_diff(self) -> bool:
...
+                line=compacted.end_line - 1,
+                start_line=compacted.start_line - 1
+                if compacted.start_line > compacted.end_line
+                else GithubObject.NotSet,
+                start_side="RIGHT" if compacted.start_line > compacted.end_line else GithubObject.NotSet,
             )
-        except GithubException:
+        except BaseException as ex:
+            logger.debug(ex)
             success = False
```
Comment (HalluxAIDev) [inline hallux/targets/github_proposal.py:131]: Catch a more specific exception or reraise the exception
```suggestion
        except Exception as ex: 
            logger.debug(ex)
            success = False


```

File: hallux/auxilary.py [hunk 10]
```diff
@@ -29,10 +29,25 @@ def set_directory(path: str | Path | TemporaryDirectory):
...
+        elif arg.startswith(name + "="):  # --key=value
+            return arg[len(name) + 1 :]
+
```
Comment (HalluxAIDev) [inline hallux/auxilary.py:52]: Remove this commented out code.
```suggestion
    for i in range(len(argv)):
        arg = argv[i]
        if arg == name:
            if len(argv) > i + 1 and not argv[i + 1].startswith("-"):
                return argv[i + 1]
        elif arg.startswith(name + "="):
            return arg[len(name) + 1 :]

```

File: hallux/targets/github_proposal.py [hunk 11]
```diff
@@ -119,10 +119,14 @@ def commit_diff(self) -> bool:
...
-                line=compacted.end_line,
-                start_line=compacted.start_line,
+                line=compacted.end_line - 1,
+                start_line=compacted.start_line - 1
+                if compacted.start_line > compacted.end_line
+                else GithubObject.NotSet,
+                start_side="RIGHT" if compacted.start_line > compacted.end_line else GithubObject.NotSet,
             )
-        except GithubException:
+        except BaseException as ex:
```
Comment (HalluxAIDev) [inline hallux/targets/github_proposal.py:128]: Catch a more specific exception or reraise the exception
```suggestion
        except GithubException as ex:

```

File: hallux/auxilary.py (hunk 10 above)
Comment (HalluxAIDev) [inline hallux/auxilary.py:52]: Remove this commented out code.
```suggestion
        if arg == name:
            if len(argv) > i + 1 and not argv[i + 1].startswith("-"):
                return argv[i + 1]
        elif arg.startswith(name + "="):
            return arg[len(name) + 1 :]

```

File: hallux/auxilary.py (hunk 1 above)
Comment (HalluxAIDev) [inline hallux/auxilary.py:50]: Remove this commented out code.
```suggestion
    for i in range(len(argv)):
        arg = argv[i]
        if arg == name:
            if len(argv) > i + 1 and not argv[i + 1].startswith("-"):
                return argv[i + 1]
        elif arg.startswith(name + "="):
```

File: hallux/targets/github_proposal.py (hunk 11 above)
Comment (HalluxAIDev) [inline hallux/targets/github_proposal.py:128]: Catch a more specific exception or reraise the exception
```suggestion
                start_side="RIGHT" if compacted.start_line < compacted.end_line else GithubObject.NotSet,
```

File: hallux/auxilary.py (hunk 1 above)
Comment (HalluxAIDev) [inline hallux/auxilary.py:50]: Remove this commented out code.
```suggestion
    for i in range(len(argv)):
        arg = argv[i]
        if arg == name:
            if len(argv) > i + 1 and not argv[i + 1].startswith("-"):
                return argv[i + 1]
        elif arg.startswith(name + "="):
```

File: hallux/targets/github_proposal.py [hunk 12]
```diff
@@ -120,9 +122,11 @@ def commit_diff(self) -> bool:
                 side="RIGHT",
                 path=str(compacted.filename),
                 line=compacted.end_line,
-                start_line=compacted.start_line,
+                start_line=compacted.start_line if compacted.start_line < compacted.end_line else GithubObject.NotSet,
+                start_side="RIGHT" if compacted.start_line < compacted.end_line else GithubObject.NotSet,
             )
-        except GithubException:
+        except BaseException as ex:
```
Comment (HalluxAIDev) [inline hallux/targets/github_proposal.py:128]: Catch a more specific exception or reraise the exception
```suggestion
        except SpecificException as ex:  # Replace BaseException with the specific exception class
```

**Review by dmitrisweb — APPROVED** [review]

**Pull request merged**